src/common/cockpitfdpassing.c

```diff
@@ -119,6 +122,15 @@ cockpit_socket_receive_fd (int  socket_fd,
                         .msg_controllen = CMSG_LEN (sizeof *out_fd) };
   assert (msg.msg_controllen <= sizeof cmsg);
 
+  /* recvmsg() has no MSG_DONTWAIT, and e.g. sudo makes stdin non-blocking with `log_output` option */
+  if (fcntl (socket_fd, F_GETFL) & O_NONBLOCK)
+    {
+      struct pollfd socket_pfd = { .fd = socket_fd, .events = POLLIN };
+      int ret = poll (&socket_pfd, 1, -1);
+      if (ret <= 0) /* error or timeout; the latter should not happen, but pass it on as "EOF" */
```

Comment:
There's a small theoretical possibility that we could get impacted here by something dumb like a `SIGWINCH`, depending on our signal dispositions as we're spawned...  can you give this the typical

```c
do
   ret = poll (&socket_pfd, 1, -1);
while (ret == -1 && errno == EINTR);
```

treatment?

Reply:
Good point, thanks! Done.